circuitpython_build_tools/build.py

```diff
@@ -64,6 +68,35 @@ def version_string(path=None, *, valid_semver=False):
 def mpy_cross(mpy_cross_filename, circuitpython_tag, quiet=False):
     if os.path.isfile(mpy_cross_filename):
         return
+
+    # Try to pull from S3
+    uname = os.uname()
+    s3_url = None
+    if uname[0] == 'Linux' and uname[4] in ('amd64', 'x86_64'):
+        s3_url = f"{S3_MPY_PREFIX}mpy-cross.static-amd64-linux-{circuitpython_tag}"
+    elif uname[0] == 'Linux' and uname[4] == 'armv7l':
+        s3_url = f"{S3_MPY_PREFIX}mpy-cross.static-raspbian-{circuitpython_tag}"
+    elif uname[0] == 'Darwin' and uname[4] == 'x86_64':
+        s3_url = f"{S3_MPY_PREFIX}mpy-cross-macos-catalina-{circuitpython_tag}"
+    else:
+        print(f"\nUnable to check S3 for sysname='{uname[0]}' release='{uname[2]}' machine='{uname[4]}'.")
+        print("  Please file an issue at https://github.com/adafruit/circuitpython-build-tools/ with the above message")
```

Comment:
I'm unsure about this text.  It'll send every M1 Mac user and every Windows user filing issues.  Instead, can it be something more like:
```python
    elif not quiet:
         print(f"Pre-built mpy-cross not available for sysname='{uname[0]}' release='{uname[2]}' machine='{uname[4]}'.")
```

(there should be binaries for 64-bit windows available, not sure why you didn't include them. inability to test?  and the darwin x86_64 binaries may work fine on m1 macs too, thanks to rosetta?)

Reply:
oh I see now you say there's some kind of permissions problem with the Windows one.  Mind filing an issue in CircuitPython with the information you have?  I tested this, but only inside of wine, not on real windows systems.  If it doesn't work, we should correct it.

Reply:
er, I guess you're saying circuitpython-build-tools doesn't work at all on Windows systems.  So the place to file the issue would be in this repo.

Reply:
Maybe. I've put no effort into figuring out if my Python environment is misconfigured.
Do you know if anyone runs `circuitpython-build-tools` from a Windows box WITHOUT using WSL2?

Reply:
my gut feeling is, no, people probably aren't using circuitpython-build-tools on windows very often, but access to a compiler would have been the worst thing and using downloaded mpy-cross.exe should cure that problem.... so it may be worth trying to make it work, when it wasn't before.